app/Filament/Resources/NewsResource.php

```diff
@@ -80,6 +83,17 @@ public static function form(Form $form): Form
                             ->label("Zdjęcie")
                             ->required()
                             ->directory(News::PHOTOS_DIRECTORY)
+                            ->saveUploadedFileUsing(function (TemporaryUploadedFile $file): string {
+                                $image = Image::read($file->getContent());
+                                $newName = "{$file->getBasename()}.webp";
+
+                                $path = News::PHOTOS_DIRECTORY . "/$newName";
+
+                                Storage::put("public/$path", $image->toWebp()->toString());
+
+                                return $path;
```

Comment:
Should not be better to create something like `Helper` to covert that files and upload? You are using the same code in three places. :thinking: